### PR TITLE
png_loader: added open from memory

### PR DESCRIPTION
**Pull request by mmaciola**
Added PngLoader::open(const char* data, uint32_t size, bool copy)
Changed variables visibility to private

### CODE REVIEW

File: src/loaders/png/tvgPngLoader.cpp [hunk 1]
```diff
@@ -51,6 +51,18 @@ bool PngLoader::open(const string& path)
     return true;
 }
 
+bool PngLoader::open(const char* data, uint32_t size, bool copy)
```
Comment (hermet) [inline src/loaders/png/tvgPngLoader.cpp:54]: Did you verify read() works fine even after data is freed?
I think it needs to copy the data if copy == true.

**Review by hermet — CHANGES_REQUESTED** [review]
Please check a comment.

File: src/loaders/png/tvgPngLoader.cpp (hunk 1 above)
Comment (mmaciola) [reply src/loaders/png/tvgPngLoader.cpp:54]: The change was tested. There is no need to copy as task aren't used for PngLoader.

Comment (hermet) [reply src/loaders/png/tvgPngLoader.cpp:54]: @mmaciola  like this sequence?
1. picture->open()
2. free(data)
3. picture->push()

Comment (mmaciola) [reply src/loaders/png/tvgPngLoader.cpp:54]: The test scenario is below. Works ok
```
    string path(EXAMPLE_DIR"/logo.png");

    //Open
    ifstream file(path);
    if (!file.is_open()) return ;
    auto size = sizeof(uint32_t) * (1000*1000);
    auto data = (char*)malloc(size);
    file.read(data, size);
    file.close();

    //Picture
    auto picture = tvg::Picture::gen();
    picture->load(data, size, false);

    //free and push
    free(data);
    canvas->push(move(picture));
```

Comment (hermet) [reply src/loaders/png/tvgPngLoader.cpp:54]: @mmaciola good thanks!

**Pull request merged**